lib/Logger/Logger.cpp

```diff
@@ -325,88 +325,278 @@ void Logger::log(char const* function, char const* file, int line,
   if (_cachedPid == 0) {
     _cachedPid = Thread::currentProcessId();
   }
-  TRI_ASSERT(_cachedPid != 0);
-  out.push_back('[');
-  StringUtils::itoa(uint64_t(_cachedPid), out);
 
-  if (_showThreadIdentifier) {
-    out.push_back('-');
-    StringUtils::itoa(uint64_t(Thread::currentThreadNumber()), out);
-  }
+  std::string out;
+  size_t offset = 0;
+
+  if (Logger::_useJson) {
+    // construct JSON output
+    StringBuffer buf;
```

Comment:
I don't like the inefficiency of this...
This creates a temporary StringBuffer object, and to copy its contents back to `out` in the end.
I know this was done to save effort (there is no `appendJsonEncoded` for std::string`), but I still don't like it overly much. Logging is called so often that we should try to make it as efficient as possible.
Because of that I also added the previously missing `if` statements that control which parts of the log information are written out to JSON. This way users can opt-in to logging certain seldomly used log parts, such as function name, line number etc., and we can reduce the size of the log output considerably.

Reply:
ok

Reply:
done